tests/tensor/rewriting/test_basic.py

```diff
@@ -1247,6 +1248,86 @@ def test_local_join_1():
     assert f.maker.fgraph.outputs[0].dtype == config.floatX
 
 
+def test_local_join_to_repeat():
+    """Test that Join(axis, x, x, ...) gets rewritten to repeat(x, n, axis)"""
+
+    # Test with vector - concatenate same vector 3 times along axis 0
+    x = vector("x")
+    s = join(0, x, x, x)
+    f = function([x], s, mode=rewrite_mode)
+
+    # Check numerical correctness
+    test_val = np.array([1.0, 2.0, 3.0], dtype=config.floatX)
+    result = f(test_val)
+    expected = np.array(
+        [1.0, 2.0, 3.0, 1.0, 2.0, 3.0, 1.0, 2.0, 3.0], dtype=config.floatX
+    )
+    assert np.allclose(result, expected)
+
+    # Check that Join was replaced with Repeat
+    ops = f.maker.fgraph.toposort()
+    assert len([n for n in ops if isinstance(n.op, Join)]) == 0
+    assert len([n for n in ops if isinstance(n.op, Repeat)]) == 1
+
+    # Test with matrix - concatenate same matrix along axis 0
+    a = matrix("a")
+    s = join(0, a, a, a, a)
+    f = function([a], s, mode=rewrite_mode)
+
+    test_mat = np.array([[1.0, 2.0], [3.0, 4.0]], dtype=config.floatX)
+    result = f(test_mat)
+    expected = np.vstack([test_mat, test_mat, test_mat, test_mat])
+    assert np.allclose(result, expected)
+
+    # Check optimization applied
+    ops = f.maker.fgraph.toposort()
+    assert len([n for n in ops if isinstance(n.op, Join)]) == 0
+    assert len([n for n in ops if isinstance(n.op, Repeat)]) == 1
```

Comment:
The RepeatOp will not actually be used. It's only used when there are vector repeats. For scalar Repeats we end up with Alloc instead (PyTensor version of BroadcastTo). Did you try to run the tests locally?